service/worker/batcher/activities.go

```diff
@@ -285,22 +431,22 @@ func startTaskProcessor(
 					})
 			case BatchTypeDelete:
 				err = processTask(ctx, limiter, task,
-					func(workflowID, runID string) error {
+					func(execution *commonpb.WorkflowExecution) error {
 						_, err := frontendClient.DeleteWorkflowExecution(ctx, &workflowservice.DeleteWorkflowExecutionRequest{
 							Namespace: batchParams.Namespace,
 							WorkflowExecution: &commonpb.WorkflowExecution{
-								WorkflowId: workflowID,
-								RunId:      runID,
+								WorkflowId: execution.WorkflowId,
+								RunId:      execution.RunId,
 							},
```

Comment:
```suggestion
							WorkflowExecution: execution,
```
and same thing repeated several more times